### PR TITLE
Fix some handle type violations.

### PR DESCRIPTION
**Pull request by jaykrell**
https://github.com/mono/mono/issues/9153

### CODE REVIEW

**Comment by jaykrell** [comment]
Should ensure is all of them, and have a way to stop new ones.

File: mono/metadata/custom-attrs.c [hunk 1]
```diff
@@ -1951,8 +1951,6 @@ mono_reflection_get_custom_attrs_by_type_handle (MonoObjectHandle obj, MonoClass
 		MONO_HANDLE_ASSIGN (result, MONO_HANDLE_NEW (MonoArray, mono_custom_attrs_construct_by_type (cinfo, attr_klass, error))); /* FIXME use coop handles for mono_custom_attrs_construct_by_type */
 		if (!cinfo->cached)
 			mono_custom_attrs_free (cinfo);
-		if (!result)
```
Comment (kumpera) [inline mono/metadata/custom-attrs.c]: Don't we still need this null check?

Comment (jaykrell) [reply mono/metadata/custom-attrs.c]: It should land at the same place w/ or w/o the check.

Comment (lambdageek) [reply mono/metadata/custom-attrs.c]: Should have been `if (MONO_HANDLE_IS_NULL (result)) goto leave`, I guess?  Or better `goto_if_nok (error, leave)`.

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -4853,7 +4853,7 @@ ves_icall_System_Reflection_Assembly_GetAotId (MonoError *error)
 {
 	char *guid = mono_runtime_get_aotid ();
 	if (guid == NULL)
-		return NULL;
+		return MONO_HANDLE_CAST (MonoString, NULL_HANDLE);
```
[the remark this answers is not in the document]
Comment (lambdageek) [reply mono/metadata/icall.c]: I think `NULL_HANDLE` didn't really work out (it never has the right type and you have to worry about someone assigning into it).  better to go with `MONO_HANDLE_NEW`

**Comment by jaykrell** [comment]
You see the larger context in `mono_reflection_get_custom_attrs_by_type_handle`, right? It is going to fall into the label? It is just goto the next line. The check does nothing.

**Comment by jaykrell** [comment]
Agreed on new vs. cast, will do.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Linux ARMv7 Interpreter

**Comment by jaykrell** [comment]
```mono_marshal_xdomain_copy_value_handle (MonoObjectHandle val, MonoError *error)
{
		MonoObjectHandle res = MONO_HANDLE_NEW (MonoObject, mono_value_box_checked (domain, klass, ((char*)val) + sizeof(MonoObject), error)); /* FIXME use handles in mono_value_box_checked */
```

**Comment by jaykrell** [comment]
@monojenkins build  Linux ARMv7 Interpreter

**Comment by jaykrell** [comment]
```mono_reflection_type_handle_mono_type:
		g_error ("Cannot handle corlib user type %s", mono_type_full_name (m_class_get_byval_arg (mono_object_class(ref))));```

**Comment by jaykrell** [comment]
@monojenkins build Linux ARMv7 Interpreter

**Comment by jaykrell** [comment]
https://github.com/mono/mono/pull/9183 contains this and is more complete.

**Pull request merged**